### DISCUSSION TITLE
unable to use webkit browser in Sauce labs due to Cypress does not provide additional launch arguments for webkit.

### DISCUSSION THREAD

**Discussion by rajenderbeeram** [cypress-io/cypress#27045] [Questions and Help]
Hi Team,

We wanted to run our tests in safari browser so we are running our cypress test scripts with Sauce labs but we are seeing page distortion and tests are not running properly(page not loading properly) due to static resolution of webkit 1280x720 for webkit browser . Sauce Labs team confirmed that due to Cypress limitation(webkit static resolution) page distortion issue not resolvable . can you please let me know when we can expect the additional arguments can be provided for webkit browser .

Thanks.
Regards,
Rajender

**Comment by portyu9** [comment]
Hi @rajenderbeeram 

There still isn't a supported WebKit equivalent to Chromium's `--window-size=...` launch argument.

Cypress's WebKit support is built on `playwright-webkit`, and the Browser Launch API currently documents command-line arguments for Chromium-based browsers and Firefox, not WebKit. A later Cypress discussion about this exact problem confirmed that Playwright WebKit does not support those window-size arguments.

If the problem is the **application viewport**, you can set that independently:

```js
export default defineConfig({
  e2e: {
    viewportWidth: 1440,
    viewportHeight: 900,
  },
})
```

or per test with:

```js
cy.viewport(1440, 900)
```

However, `viewportWidth` / `viewportHeight` change the application's viewport; they do **not** change the actual WebKit browser/window or virtual display resolution. If Sauce Labs requires the underlying WebKit surface itself to be larger than `1280x720`, there is no Cypress launch-argument workaround for that today.

Also note that Cypress WebKit is still experimental and uses the WebKit engine via Playwright; it is not the actual Safari browser.

Please mark this answer as accepted if it helped, thank you.
